### PR TITLE
Update to .NET 7 SDKs

### PR DESCRIPTION
**Pull request by mattleibow**
Bump to:

* .NET SDK 7.0.100-preview.6.22276.1
* .NET Runtime 7.0.0-preview.5.22272.3
* Android 33.0.0-preview.4.25
* iOS/MacCatalyst/tvOS 15.4.704-ci.net7-0
* macOS 12.3.704-ci.net7-0
* Tizen 7.0.100-preview.6.7

Other changes:

* Build with `dotnet build`
* Use `net7.0` target framework
* `NET6_0` -> `NET6_0_OR_GREATER`
* Provision Android API 32
* .NET 7 workload packs need `7.0.100-preview.6` version band
* `Image*UnitTest` needs a `TaskCompletionSource` in .NET 7
* Tizen packs should check `$(IncludeTizenTargetFrameworks)`

Workarounds:

* Usage of `new NSUrlRequest(uri!)` to avoid NRT warnings
* `GetScrollYOnGenList()` on Tizen needed new linker warnings ignored
* `$(NoWarn)` is needed to workaround linker analyzers:

https://github.com/dotnet/linker/issues/2817

* Added additional logging for `XamlC` failing on:

```
    Task XamlCTask
    Error XamlApp.xaml Invalid URI: The hostname could not be parsed. 
       at System.Uri.CreateThis(String uri, Boolean dontEscape, UriKind uriKind, UriCreationOptions& creationOptions)
       at System.Uri..ctor(String uriString, UriKind uriKind)
       at Microsoft.Maui.Controls.ResourceDictionary.RDSourceTypeConverter.GetResourcePath(Uri uri, String rootTargetPath)
       at Microsoft.Maui.Controls.XamlC.RDSourceTypeConverter.ConvertFromString(String value, ILContext context, BaseNode node)+MoveNext()
       at Microsoft.Maui.Controls.Build.Tasks.NodeILExtensions.PushConvertedValue(ValueNode node, ILContext context, TypeReference targetTypeRef, TypeReference typeConverter, IEnumerable`1 pushServiceProvider, Boolean boxValueTypes, Boolean unboxValueTypes)+MoveNext()
       at Microsoft.Maui.Controls.Build.Tasks.SetPropertiesVisitor.Set(VariableDefinition parent, String localName, INode node, IXmlLineInfo iXmlLineInfo, ILContext context)+MoveNext()
       at Microsoft.Maui.Controls.Build.Tasks.ILProcessorExtensions.Append(ILProcessor processor, IEnumerable`1 instructions)
       at Microsoft.Maui.Controls.Build.Tasks.SetPropertiesVisitor.Visit(ValueNode node, INode parentNode)
       at Microsoft.Maui.Controls.Xaml.ElementNode.Accept(IXamlNodeVisitor visitor, INode parentNode)
       at Microsoft.Maui.Controls.Build.Tasks.SetResourcesVisitor.Visit(ElementNode node, INode parentNode)
       at Microsoft.Maui.Controls.Xaml.ElementNode.Accept(IXamlNodeVisitor visitor, INode parentNode)
       at Microsoft.Maui.Controls.Xaml.ElementNode.Accept(IXamlNodeVisitor visitor, INode parentNode)
       at Microsoft.Maui.Controls.Xaml.RootNode.Accept(IXamlNodeVisitor visitor, INode parentNode)
       at Microsoft.Maui.Controls.Build.Tasks.XamlCTask.TryCoreCompile(MethodDefinition initComp, ILRootNode rootnode, String xamlFilePath, Exception& exception)
```

### CODE REVIEW

File: src/Controls/src/Build.Tasks/CompiledConverters/RDSourceTypeConverter.cs [hunk 1]
```diff
@@ -40,9 +40,17 @@ public IEnumerable<Instruction> ConvertFromString(string value, ILContext contex
...
-			var resourcePath = ResourceDictionary.RDSourceTypeConverter.GetResourcePath(uri, rootTargetPath);
+			var uri = new Uri(value, UriKind.Relative);
+			string resourcePath;
+			try
+			{
+				resourcePath = ResourceDictionary.RDSourceTypeConverter.GetResourcePath(uri, rootTargetPath);
+			}
+			catch (Exception exc)
+			{
+				throw new BuildException(BuildExceptionCode.GetResourcePath, node, exc, uri, rootTargetPath);
```
Comment (jonathanpeppers) [inline src/Controls/src/Build.Tasks/CompiledConverters/RDSourceTypeConverter.cs:52]: This was failing occasionally with:
```
Task XamlCTask
Error XamlApp.xaml Invalid URI: The hostname could not be parsed. 
   at System.Uri.CreateThis(String uri, Boolean dontEscape, UriKind uriKind, UriCreationOptions& creationOptions)
   at System.Uri..ctor(String uriString, UriKind uriKind)
   at Microsoft.Maui.Controls.ResourceDictionary.RDSourceTypeConverter.GetResourcePath(Uri uri, String rootTargetPath)
   at Microsoft.Maui.Controls.XamlC.RDSourceTypeConverter.ConvertFromString(String value, ILContext context, BaseNode node)+MoveNext()
   at Microsoft.Maui.Controls.Build.Tasks.NodeILExtensions.PushConvertedValue(ValueNode node, ILContext context, TypeReference targetTypeRef, TypeReference typeConverter, IEnumerable`1 pushServiceProvider, Boolean boxValueTypes, Boolean unboxValueTypes)+MoveNext()
   at Microsoft.Maui.Controls.Build.Tasks.SetPropertiesVisitor.Set(VariableDefinition parent, String localName, INode node, IXmlLineInfo iXmlLineInfo, ILContext context)+MoveNext()
   at Microsoft.Maui.Controls.Build.Tasks.ILProcessorExtensions.Append(ILProcessor processor, IEnumerable`1 instructions)
   at Microsoft.Maui.Controls.Build.Tasks.SetPropertiesVisitor.Visit(ValueNode node, INode parentNode)
   at Microsoft.Maui.Controls.Xaml.ElementNode.Accept(IXamlNodeVisitor visitor, INode parentNode)
   at Microsoft.Maui.Controls.Build.Tasks.SetResourcesVisitor.Visit(ElementNode node, INode parentNode)
   at Microsoft.Maui.Controls.Xaml.ElementNode.Accept(IXamlNodeVisitor visitor, INode parentNode)
   at Microsoft.Maui.Controls.Xaml.ElementNode.Accept(IXamlNodeVisitor visitor, INode parentNode)
   at Microsoft.Maui.Controls.Xaml.RootNode.Accept(IXamlNodeVisitor visitor, INode parentNode)
   at Microsoft.Maui.Controls.Build.Tasks.XamlCTask.TryCoreCompile(MethodDefinition initComp, ILRootNode rootnode, String xamlFilePath, Exception& exception)
```
It didn't happen after adding more logging... Maybe it's random?

File: eng/pipelines/handlers.yml [hunk 2]
```diff
@@ -75,14 +75,14 @@ parameters:
         vmImage: $(windowsNet6VmImage)
         bootsAndroid: $(Android.Msi)
         bootsiOS: $(iOS.Msi)
-        artifact: net6-windows
+        artifact: net7-windows
```
Comment (mattleibow) [inline eng/pipelines/handlers.yml:78]: Probably should just remove version numbers here.

**Review by mattleibow — COMMENTED** [review]
LGTM

File: eng/pipelines/handlers.yml (hunk 2 above)
Comment (jonathanpeppers) [reply eng/pipelines/handlers.yml:78]: I'm going to put `build-windows` and `build-macos`, I think.

**Pull request merged**